### PR TITLE
build: added a github workflow to verify contributors commits

### PR DESCRIPTION
**Pull request by etorreborre**
This is an attempt to automate the manual consistency checks currently done when a new contributor is added to the list of contributors. We check that:

 - there are no merge commits in the PR
 - the PR commits are signed
 - for the commit adding a contributor we check that
    - the agreement says 'yes' 
    - the commit author name is provided in the 'Github User'  column
    - the commit author full name is provided in the 'Name' column
    - the commit author email address is one of the addresses provided in the 'Emails' column

### CODE REVIEW

**Comment by etorreborre** [comment]
@mrinalwadhwa this is an attempt at verifying contributors commits. Now the question is: how I can test that this workflow action is working correctly :-)? I ran some local tests on the main script and they worked fine but I wouldn't be surprised if there were all sorts of edge cases. What do you think?

File: .github/workflows/all.yml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            This workflow checks that all commits follow the Ockam Commit Message Convention
+            https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages
+
+            We check all commits from HEAD backwards till the commit with commit hash: ${FIRST_COMMIT}.
+
+            ERROR:
+            For this to work the commit with commit hash: ${FIRST_COMMIT} should be an ancestor of HEAD
+            but it seems this is not the case with the current HEAD.
+
+            Try rebasing to the develop branch of ockam.
```
Comment (shanesveller) [inline .github/workflows/all.yml:72]:
```suggestion
            Try rebasing to the develop branch of ockam-contributors.
```

File: .github/workflows/all.yml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+              Please rebase to main.
+            " && exit 1
+          fi
+
+      - name: Check all commits in are Verified by Github (Pull Request)
+        if: github.event_name == 'pull_request'
+        env:
+          PR_SENDER: ${{ github.event.pull_request.user.login }}
+        run: |
+          unverified=$(cat commits.json | jq --raw-output '.[] | [.sha, .commit.verification.verified] | @csv' | grep false || echo '')
```
Comment (shanesveller) [inline .github/workflows/all.yml:91]: 👍🏻  This helps me understand why the previous step at ~L46 wasn't based on `git log` or similar. I think you could potentially also do something like:

`jq --raw-output '.[] | select(.commit.verification.verified == false) | .sha'`

 if that's any clearer or otherwise helpful?
https://stedolan.github.io/jq/manual/#Builtinoperatorsandfunctions

File: .github/workflows/all.yml (hunk 1 above)
Comment (etorreborre) [reply .github/workflows/all.yml:72]: Thanks

**Comment by mrinalwadhwa** [comment]
@metaclips has some approaches figured out to test workflows before merge .. I think he usually does this in a fork.

File: .github/workflows/all.yml (hunk 2 above)
Comment (etorreborre) [reply .github/workflows/all.yml]: Yes that is indeed better, thanks!
